--- home/sovol/klipper/klippy/extras/filament_switch_sensor.py	2026-08-29 13:42:13.000000000 +0000
+++ DEMON/DEMON_SV08_MAX_REPLACEMENT_FILES/filament_switch_sensor.py	2026-08-29 14:20:38.086409000 +0000
@@ -63,15 +63,20 @@
         if is_filament_present == self.filament_present:
             return
         self.filament_present = is_filament_present
-        if not is_filament_present:
-            self.gcode.run_script_from_command('SET_PIN PIN=green_led VALUE=0.00')
-            self.gcode.run_script_from_command('SET_GCODE_VARIABLE MACRO=variables VARIABLE=stop_feeding VALUE=True')
         eventtime = self.reactor.monotonic()
         if eventtime < self.min_event_systime or not self.sensor_enabled:
             # do not process during the initialization time, duplicates,
             # during the event delay time, while an event is running, or
             # when the sensor is disabled
             return
+        if not is_filament_present:
+            self.gcode.run_script_from_command('SET_PIN PIN=LED_Green VALUE=0.00')
+            self.gcode.run_script_from_command('SET_GCODE_VARIABLE MACRO=variables VARIABLE=stop_feeding VALUE=True')
+        else:
+            self.gcode.run_script_from_command('SET_PIN PIN=LED_Green VALUE=1.00')
+            self.gcode.run_script_from_command('SET_GCODE_VARIABLE MACRO=variables VARIABLE=stop_feeding VALUE=False')
+            self.gcode.run_script_from_command('_AUTO_FEED_MESSAGE')
+            self.gcode.run_script_from_command('FEED_LOOP') ##############################################################
         # Determine "printing" status
         idle_timeout = self.printer.lookup_object("idle_timeout")
         is_printing = idle_timeout.get_status(eventtime)["state"] == "Printing"
