--- home/sovol/klipper/klippy/extras/z_offset_calibration.py	2026-08-29 13:42:13.000000000 +0000
+++ DEMON/DEMON_SV08_MAX_REPLACEMENT_FILES/z_offset_calibration.py	2026-08-29 14:20:38.853807200 +0000
@@ -3,6 +3,7 @@
 # Copyright (C) 2024-2025 Sovol3d <info@sovol3d.com>
 #
 # This file may be distributed under the terms of the GNU GPLv3 license.
+from random import randint
 from . import probe, probe_eddy_current, manual_probe
 import math
 import configparser
@@ -13,7 +14,7 @@
         self.config = config
         x_pos_endstop, y_pos_endstop = config.getfloatlist("endstop_xy_position", count=2)
         self.endstop_x_pos, self.endstop_y_pos = x_pos_endstop, y_pos_endstop
-        self.z_hop = config.getfloat("z_hop", default=2.0)
+        self.z_hop = config.getfloat("z_hop", default=5.0)
         self.z_hop_speed = config.getfloat('z_hop_speed', 5., above=0.)
         self.zconfig = config.getsection('stepper_z')
         self.endstop_pin = self.zconfig.get('endstop_pin')
@@ -72,7 +73,8 @@
         self.set_z_offset(offset=0.)
         # Move to probe position
         gcmd.respond_info("ZoffsetCalibration: Toolhead move ...")
-        self.toolhead.manual_move([self.endstop_x_pos, self.endstop_y_pos], self.speed)
+        self.toolhead.manual_move([(self.endstop_x_pos + generator()), (self.endstop_y_pos + generator())], self.speed)
+        # self.toolhead.manual_move([self.endstop_x_pos, self.endstop_y_pos], self.speed)
         # Contact probe calibration
         gcmd.respond_info("ZoffsetCalibration: Toolhead probing ...")
         zendstop_p = _contact_probe.run_contact_probe(gcmd)
@@ -94,7 +96,7 @@
             zendstop_p1 = _contact_probe.run_contact_probe(gcmd)
             diff_z = abs(zendstop_p1[2] - zendstop_p[2])
             zendstop_p = zendstop_p1
-            if diff_z <= 0.0125:
+            if diff_z <= 0.02:
                 gcmd.respond_info("ZoffsetCalibration: Toolhead check success.")
                 break
             reprobe_cnt += 1
@@ -115,3 +117,6 @@
     
 def load_config(config):
     return ZoffsetCalibration(config)
+
+def generator():
+    return randint(-50, 50)
\ No newline at end of file
